tests/cpu-compatibility.tftest.hcl

```diff
@@ -0,0 +1,104 @@
+variables {
+  vpc_id              = "vpc-12345678"
+  subnet_ids          = ["subnet-12345678", "subnet-87654321"]
+  stage               = "test"
+  namespace           = "mp"
+  name                = "ssm-agent"
+  region              = "us-east-1"
+  availability_zones  = ["us-east-1a"]
+  nat_gateway_enabled = true
+  ipv6_enabled        = true
+}
+
+### TESTING INSTANCE and ARCHITECTURE COMPATIBILITY ###
+# https://docs.aws.amazon.com/ec2/latest/instancetypes/instance-type-names.html
+# https://aws.amazon.com/ec2/instance-types/
+
+# Test valid x86_64 instance type
+run "valid_x86_64_instance" {
+  command = plan
+
+  variables {
```

Comment:
I think we should improve the readability by moving this into a helper module where we can use loops and locals like this:
```hcl
locals {
  instances = [
    {
      instance_type = "t3.micro"
      architecture  = "x86_64"
      expected      = true
    },
    {
      instance_type = "t4g.micro"
      architecture  = "arm64"
      expected      = true
    },
    {
      instance_type = "t4g.micro"
      architecture  = "x86_64"
      expected      = false
    },
    {
      instance_type = "t3.micro"
      architecture  = "arm64"
      expected      = false
    },
    {
      instance_type = "g3s.xlarge"
      architecture  = "arm64"
      expected      = false
    },
    {
      instance_type = "g4dn.xlarge"
      architecture  = "x86_64"
      expected      = true
    }
  ]
}
```
Unfortunately, for_each is not supported yet for tests, but we can workaround that with `run.module` block. 
How does that sound?

Reply:
Maybe we can use the `setup` module just to build this map of instance types, architecture, and results, similar to how you generate a random ID for the test? In this case it's fine to apply, it won't create any AWS resources.

TBH, I need to dig deeper into test best practices and patterns. My only concern here now is that we have tons of repeatable pieces of code. I wonder what are the best ways to handle that fare or the native test framework.

Reply:
I tried looking around too for see how others do it and/or best practices, but since it's a relatively new thing, there's not much out there on it haha.

> Maybe we can use the setup module just to build this map of instance types, architecture, and results,
if we build this map, we still can't use any loops in the `.tftest`, so we'd still have to reference it like `run.setup.cpuCase1.instance_type`, and then at that point it might be an additional necessary layer.